client/src/pages/CreatedConcert.js

```diff
@@ -16,60 +16,176 @@ import { ButtonFull } from "../components/buttons/ButtonFull";
 import { ButtonContainer } from "../components/buttons/ButtonContainer";
 import { DetailTextContainer } from "../components/cards/DetailTextContainer";
 import UserCard from "../components/User/UserCard";
+import { EditTitle } from "../components/forms/EditTitle";
+import { EditInput } from "../components/forms/EditInput";
+import EditArrivalDropdown from "../components/forms/EditArrivalDropdown";
+import { EditTextarea } from "../components/forms/EditTextarea";
 import useGetConcert from "../hooks/useGetConcert";
+import { patchConcert } from "../api/concerts";
 
 export default function CreateConcert() {
   const { concertId } = useParams();
-  const { concert, isLoading, errorMessage } = useGetConcert(concertId);
+  const { concert, isLoading } = useGetConcert(concertId);
+  const [edit, setEdit] = React.useState(false);
+  const [updatedArtist, setUpdatedArtist] = React.useState("");
+  const [updatedDate, setUpdatedDate] = React.useState("");
+  const [updatedArrival, setUpdatedArrival] = React.useState("");
+  const [updatedLocation, setUpdatedLocation] = React.useState("");
+  const [updatedCity, setUpdatedCity] = React.useState("");
+  const [updatedPrice, setUpdatedPrice] = React.useState("");
+  const [updatedDetailText, setUpdatedDetailText] = React.useState("");
 
-  if (errorMessage) {
-    return <div>{errorMessage}</div>;
-  }
-  if (isLoading) {
-    return <div>Is loading...</div>;
-  }
+  // HANDLE EDIT
+  const handleEditClick = (event) => {
+    event.preventDefault();
+
+    //
+    setUpdatedArtist(concert.artist);
+    setUpdatedDate(concert.date);
+    setUpdatedArrival(concert.arrival);
+    setUpdatedLocation(concert.location);
+    setUpdatedCity(concert.city);
+    setUpdatedPrice(concert.price);
+    setUpdatedDetailText(concert.detailText);
+
+    //
+    setEdit(true);
+  };
+
+  // SAVE EDIT
+
+  const handleSaveClick = () => {
+    const updatedConcert = {
+      artist: updatedArtist,
+      date: updatedDate,
+      arrival: updatedArrival,
+      location: updatedLocation,
+      city: updatedCity,
+      price: updatedPrice,
+      detailText: updatedDetailText,
+    };
+
+    patchConcert(concertId, updatedConcert);
+
```

Comment:
In order to display the changes, you need to update `{concert}` here somehow.  So far you're only patching them in the DB, that's why after a refresh you'll see the changes here because `useGetConcert` is fired again. Maybe include a refresh function in your hook

Reply:
Oh, thank you! I add a little refresh function and it works :) 
Of course I'll refactor the handleSaveClick later...